File: lib/models/canteen.dart
class Canteen {
  final String id;
  final String name;
  final String location;
  final String image;
  final String description;
  final String operatingHours;
  final double rating;
  final List<String> categories;
  final double latitude;
  final double longitude;
  final List<String> menu;

  Canteen({
    required this.id,
    required this.name,
    required this.location,
    required this.image,
    required this.description,
    required this.operatingHours,
    this.rating = 0.0,
    this.categories = const [],
    required this.latitude,
    required this.longitude,
    required this.menu,
  });

  factory Canteen.fromJson(Map<String, dynamic> json) {
    return Canteen(
      id: json['id'] as String,
      name: json['name'] as String,
      location: json['location'] as String,
      image: json['image'] as String,
      description: json['description'] as String,
      operatingHours: json['operatingHours'] as String,
      rating: (json['rating'] as num?)?.toDouble() ?? 0.0,
      categories: (json['categories'] as List<dynamic>?)?.cast<String>() ?? [],
      latitude: (json['latitude'] as num).toDouble(),
      longitude: (json['longitude'] as num).toDouble(),
      menu: (json['menu'] as List<dynamic>?)?.cast<String>() ?? [],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'location': location,
      'image': image,
      'description': description,
      'operatingHours': operatingHours,
      'rating': rating,
      'categories': categories,
      'latitude': latitude,
      'longitude': longitude,
      'menu': menu,
    };
  }
} 

File: src/main/java/com/example/demo/entity/Canteen.java
package com.example.demo.entity;

import java.util.List;

import jakarta.persistence.ElementCollection;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;

@Entity
public class Canteen {
    @Id
    private String id;
    private String name;
    private String location;
    private String image;
    private String description;
    private String operatingHours;
    private double rating;
    @ElementCollection
    private List<String> categories;
    private double latitude;
    private double longitude;
    @ElementCollection
    private List<String> menu;

    // Getter/Setter 省略，为简洁可用 Lombok 或 IDE 自动生成
    public String getId() { return id; }
    public void setId(String id) { this.id = id; }
    public String getName() { return name; }
    public void setName(String name) { this.name = name; }
    public String getLocation() { return location; }
    public void setLocation(String location) { this.location = location; }
    public String getImage() { return image; }
    public void setImage(String image) { this.image = image; }
    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }
    public String getOperatingHours() { return operatingHours; }
    public void setOperatingHours(String operatingHours) { this.operatingHours = operatingHours; }
    public double getRating() { return rating; }
    public void setRating(double rating) { this.rating = rating; }
    public List<String> getCategories() { return categories; }
    public void setCategories(List<String> categories) { this.categories = categories; }
    public double getLatitude() { return latitude; }
    public void setLatitude(double latitude) { this.latitude = latitude; }
    public double getLongitude() { return longitude; }
    public void setLongitude(double longitude) { this.longitude = longitude; }
    public List<String> getMenu() { return menu; }
    public void setMenu(List<String> menu) { this.menu = menu; }
}